src/i2c.rs

```diff
@@ -220,12 +220,16 @@ macro_rules! wait_for_flag {
         let sr1 = $i2c.sr1.read();
 
         if sr1.berr().bit_is_set() {
+            $i2c.sr1.write(|w| w.berr().clear_bit());
             Err(Other(Error::Bus))
         } else if sr1.arlo().bit_is_set() {
+            $i2c.sr1.write(|w| w.arlo().clear_bit());
             Err(Other(Error::Arbitration))
         } else if sr1.af().bit_is_set() {
+            $i2c.sr1.write(|w| w.af().clear_bit());
             Err(Other(Error::Acknowledge))
         } else if sr1.ovr().bit_is_set() {
+            $i2c.sr1.write(|w| w.ovr().clear_bit());
```

Comment:
Hmm, is `write` correct here? These look like no-op to me.

Reply:
I could be wrong, but I believe these are there to reset the flags

Reply:
Yes, but `sr1.write(|w| w.any_field().clear_bit());` basically writes 0 to the whole register, regardless of the field name. That's why this looks like a mistake.

Reply:
Oh right, I guess that should be `modify` instead. Good catch, I'll go ahead and fix that